ome/datasource_groupdevices.go

```diff
@@ -119,22 +126,27 @@ func (g groupDevicesDataSource) Read(ctx context.Context, req tfsdk.ReadDataSour
 	devices = omeClient.GetUniqueDevices(devices)
 
 	for _, device := range devices {
-		devIDs = append(devIDs, types.Int64{Value: device.ID})
-		devSvcTags = append(devSvcTags, types.String{Value: device.DeviceServiceTag})
+		devIDs = append(devIDs, types.Int64Value(device.ID))
+		devSvcTags = append(devSvcTags, types.StringValue(device.DeviceServiceTag))
 	}
 
-	devIDsTfsdk := types.List{
-		ElemType: types.Int64Type,
-	}
-	devIDsTfsdk.Elems = devIDs
-	groupDevices.DeviceIDs = devIDsTfsdk
+	devIDsTfsdk, _ := types.ListValue(
+		types.Int64Type,
+		devIDs,
+	)
 
-	devSTsTfsdk := types.List{
-		ElemType: types.StringType,
+	if !devIDsTfsdk.IsUnknown() {
+		groupDevices.DeviceIDs = devIDsTfsdk
 	}
-	devSTsTfsdk.Elems = devSvcTags
-	groupDevices.DeviceServicetags = devSTsTfsdk
 
+	devSTsTfsdk, _ := types.ListValue(
+		types.StringType,
+		devSvcTags,
+	)
+
+	if !devSTsTfsdk.IsUnknown() {
+		groupDevices.DeviceServicetags = devSTsTfsdk
+	}
```

Comment:
1. Errors should not be ignored unless there is a specific reason.
2. The check for unknown is redundant. types.ListValue() always returns a known list.

Reply:
unsetting the unknown after testing the datasource and resource.